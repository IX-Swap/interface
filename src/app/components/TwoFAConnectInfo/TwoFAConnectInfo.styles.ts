import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  wrapper: {
    padding: theme.spacing(5),
    borderRadius: 8
  },
  infoBlock: {
    margin: theme.spacing(0.5, 2.5, 2.5, 0)
  },
  button: {
    width: '100%'
  }
}))
