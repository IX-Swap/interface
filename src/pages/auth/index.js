import React, { Suspense } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import SignIn from './pages/sign-in';
import Reset from './pages/reset';

const Confirm = React.lazy(() => import('./pages/confirm'));

export default function Auth() {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      <Switch>
        <Route path="/auth/sign-in" exact component={SignIn} />
        <Route path="/auth/reset" exact component={Reset} />
        <Route path="/auth/confirm" exact component={Confirm} />

        <Route path="/auth" render={() => <Redirect to="/auth/sign-in" />} />
      </Switch>
    </Suspense>
  );
}
