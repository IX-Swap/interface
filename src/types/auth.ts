export type BaseArgs = Record<string, any>

export interface LoginArgs extends BaseArgs {
  email: string
  password: string
  otp?: string
}

export interface SignupArgs extends BaseArgs {
  email: string
  name?: string
  password: string
}

export interface VerifySignupArgs extends BaseArgs {
  verificationToken: string
}

export interface RequestPasswordResetArgs extends BaseArgs {
  email: string
}

export interface ResendVerificationEmailArgs extends BaseArgs {
  email: string
}

export interface CompletePasswordResetArgs {
  email: string
  resetToken: string
  newPassword: string
  tenantId?: string
}

export interface GetUserArgs extends BaseArgs {
  userId: string
}

export const MAX_LOGIN_ATTEMPTS = 3
export const LOCK_LOGIN_ERROR_CODES = ['RECO-RLE291', 'RWC0-70531O']
