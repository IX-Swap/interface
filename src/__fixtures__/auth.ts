import { user } from './user'
import { APIServiceResponse } from 'v2/services/api/types'
import User from 'v2/types/user'
import {
  CompletePasswordResetArgs,
  LoginArgs,
  RequestPasswordResetArgs,
  SignupArgs,
  VerifySignupArgs
} from 'v2/types/auth'

const email = 'alex@investax.io'
const password = 'Qwerty12345_'

export const authURLs = {
  login: '/auth/sign-in',
  signup: '/auth/registrations',
  verifySignup: '/auth/registrations/confirm',
  getUser: '/auth/profiles',
  requestPasswordReset: '/auth/password/reset/start',
  completePasswordReset: '/auth/password/reset/confirm'
}

export const loginArgs: LoginArgs = {
  otp: '123456',
  password,
  email
}

export const signupArgs: SignupArgs = {
  name: 'Alex Solovev',
  email,
  password
}

export const verifySignupArgs: VerifySignupArgs = {
  verificationToken: 'mytoken'
}

export const requestPasswordResetArgs: RequestPasswordResetArgs = {
  email
}

export const completePasswordResetArgs: CompletePasswordResetArgs = {
  email,
  resetToken: 'token',
  newPassword: password
}

export const loginResponseSuccess: APIServiceResponse<User> = {
  data: user,
  message: 'Welcome',
  success: true
}

export const loginResponseFailure: APIServiceResponse<User> = {
  message: 'error',
  success: false
}

export const signupResponseSuccess: APIServiceResponse = {
  message: 'A verification e-mail has been sent to your account',
  success: true
}

export const signupResponseFailure: APIServiceResponse = {
  message: 'Sorry but this email address is already taken',
  success: false
}

export const verifySignupResponseSuccess: APIServiceResponse = {
  message: 'Your account has been verified',
  success: true
}

export const verifySignupResponseFailure: APIServiceResponse = {
  message: 'The token has already been verified, has expired, or is invalid',
  success: false
}

export const getUserResponseSuccess: APIServiceResponse<User> = {
  data: user,
  message: 'OK',
  success: true
}

export const getUserResponseFailure: APIServiceResponse<User> = {
  message: 'Access denied',
  success: false
}

export const requestPasswordResetResponseSuccess: APIServiceResponse = {
  message: 'OK',
  success: true
}

export const requestPasswordResetResponseFailure: APIServiceResponse = {
  message: 'Error',
  success: false
}

export const completePasswordResetResponseSuccess: APIServiceResponse = {
  message: 'Success',
  success: true
}

export const completePasswordResetResponseFailure: APIServiceResponse = {
  message: 'Reset token is already used, expired, or invalid',
  success: false
}
