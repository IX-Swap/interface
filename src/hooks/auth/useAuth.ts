import { useCachedUser } from 'hooks/auth/useCachedUser'

export const useAuth = () => {
  const user = useCachedUser()

  return {
    isAuthenticated: true,
    user: user
  }
}

export default useAuth
