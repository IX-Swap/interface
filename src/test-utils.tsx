import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
