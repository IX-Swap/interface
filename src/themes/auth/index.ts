import { createTheme } from '@mui/material/styles'
import { darkTheme } from 'themes/auth/dark'
import { getThemeOverrides } from 'themes/auth/overrides'
import { typography } from 'themes/auth/typography'

export enum AppTheme {
  Dark = 'Dark'
}

export const getAppTheme = () => {
  const theme = createTheme({ ...darkTheme, typography })

  theme.components = getThemeOverrides(theme)

  return theme
}
