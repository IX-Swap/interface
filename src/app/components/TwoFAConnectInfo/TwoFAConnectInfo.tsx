import React from 'react'
import { Button, Grid, Paper, Typography } from '@mui/material'
import { useStyles } from './TwoFAConnectInfo.styles'
import { useAuth } from 'hooks/auth/useAuth'
import { Icon } from 'ui/Icons/Icon'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { SecurityRoute } from 'app/pages/security/router/config'

export const TwoFAConnectInfo = () => {
  const { user = { enable2Fa: undefined } } = useAuth()
  const { enable2Fa } = user
  const classes = useStyles()

  if (enable2Fa === true) {
    return null
  }

  return (
    <Paper className={classes.wrapper}>
      <Grid container direction={'column'}>
        <Grid
          item
          container
          alignItems={'center'}
          justifyContent={'space-between'}
        >
          <Grid item>
            <Typography variant={'h5'}>2FA</Typography>
          </Grid>
          <Grid item>
            <Icon name='security' />
          </Grid>
        </Grid>
        <Grid item className={classes.infoBlock}>
          <Typography variant={'body1'}>
            Secure your account with two-factor authentication
          </Typography>
        </Grid>
        <Grid item>
          <Button
            variant={'outlined'}
            component={AppRouterLinkComponent}
            to={SecurityRoute.setup2fa}
            className={classes.button}
          >
            Enable 2FA
          </Button>
        </Grid>
      </Grid>
    </Paper>
  )
}
