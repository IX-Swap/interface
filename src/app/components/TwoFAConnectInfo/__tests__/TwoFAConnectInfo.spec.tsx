import React from 'react'
import { render } from 'test-utils'
import { TwoFAConnectInfo } from 'app/components/TwoFAConnectInfo/TwoFAConnectInfo'
import * as useAuth from 'hooks/auth/useAuth'

describe('TwoFAConnectInfo', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('renders null if enable2Fa is true', () => {
    const objResponse = {
      user: {
        enable2Fa: true
      }
    }

    jest.spyOn(useAuth, 'useAuth').mockImplementation(() => objResponse as any)

    const { container } = render(<TwoFAConnectInfo />)
    expect(container).toBeEmptyDOMElement()
  })

  it('should match snapshot when enable2Fa is not true', () => {
    const objResponse = {
      user: {
        enable2Fa: false
      }
    }

    jest.spyOn(useAuth, 'useAuth').mockImplementation(() => objResponse as any)

    const { container } = render(<TwoFAConnectInfo />)
    expect(container).toMatchSnapshot()
  })
})
