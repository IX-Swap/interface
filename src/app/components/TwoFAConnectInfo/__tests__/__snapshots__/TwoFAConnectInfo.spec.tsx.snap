// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoFAConnectInfo should match snapshot when enable2Fa is not true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 makeStyles-wrapper-4 css-11gelwl-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-1gb55do-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-14eqi03-MuiTypography-root"
          >
            2FA
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <svg
            class="css-1s8o7ec"
            fill="#778194"
            height="24px"
            width="24px"
          >
            <use
              href="image-mock#security"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item makeStyles-infoBlock-5 css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
        >
          Secure your account with two-factor authentication
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root makeStyles-button-6 css-119vr5z-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
          href="/app/settings/setup-2fa"
          tabindex="0"
        >
          Enable 2FA
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
